src/main/java/com/conveyal/gtfs/model/Entity.java

```diff
@@ -49,11 +49,27 @@
 
     private static final long serialVersionUID = -3576441868127607448L;
     public static final int INT_MISSING = Integer.MIN_VALUE;
-    public long sourceFileLine;
+    public int sourceFileLine;
 
     /* The feed from which this entity was loaded. */
     transient GTFSFeed feed;
 
+    /**
+     * @return a key that according to the GTFS spec should uniquely identify this entity, either alone or together
+     *          with a sequence number. For example stop_times and shapes have no single field that uniquely
+     *          identifies a row, and the stop_sequence or shape_pt_sequence must also be considered.
+     */
+    public String getId () {
+        return null;
+    }
+
+    /**
+     * @return the integer second element of a compound key, for those elements that require one.
+     */
+    public Integer getSequenceNumber () {
+        return null;
```

Comment:
Looks like a WIP, but just leaving a comment to make sure this gets implemented or removed.

Reply:
As above, already used. I added a comment explaining it's overridden on subclasses. For example, on stop times:

```
@Override
    public Integer getSequenceNumber() {
        return stop_sequence; // Compound key of StopTime is (trip_id, stop_sequence)
    }

```